blockmanager.go

```diff
@@ -425,16 +429,20 @@ func (b *blockManager) handleDonePeerMsg(peers *list.List, p *peer) {
 
 	// Remove requested transactions from the global map so that they will
 	// be fetched from elsewhere next time we get an inv.
-	for k := range p.requestedTxns {
-		delete(b.requestedTxns, k)
+	for k, txPeer := range b.requestedTxns {
```

Comment:
I think the structure should categorize the requests by peer.  The intent for most of these is to be able to query information based on a peer and looping everything for every peer to see if the peer matches is quite inefficient.

Perhaps something like the following where int64 is the unique peer ID.
```Go
requestedTxns:   make(map[int64]map[wire.ShaHash]struct{})
```

Then adding a peer would be something like  `requestedTxns[p.ID()] = make(map[wire.ShaHash]struct{})` and looping it here would be a simple matter of: `for k := range b.requestedTxns[p.ID()] {`.